### PR TITLE
feat!: :sparkles: added bigquery subcription capability

### PR DESCRIPTION
**Pull request by rs1986x**
Hi! I wanted to integrate the recently released push to bigquery functionality from google provider 4.32.

I tested as thoroughly as possible on our dev setup, i have to say though i was not able to run the kitchen tests, i can't follow the instructions to the letter give some accounts and privileges constraint that i have. I tried to run it against an already existing project but i was not able to, maybe you can help with that.

Thanks for the consideration, any feedback is highly appreciated

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
Checking for missing newline at end of file
Error: No newline at end of file ./examples/bigquery/main.tf
 ```

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by rs1986x** [comment]
let's put it in stand by, i noticed an issue with the resource creation and raised https://github.com/hashicorp/terraform-provider-google/issues/12332

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by rs1986x** [comment]
Hello! Could somebody share the output of the failed cicd so that i may fix it please? 
P.S. i have been using the module for a while now with no problems

**Comment by apeabody** [comment]
Hi @rs1986x - Thanks for the contribution!  The CI error appeared to be unrelated, so I just triggered a re-run.  Cheers!

**Comment by apeabody** [comment]
Hi @rs1986x - I haven't checked if this is related to your change (duplicate pubsub topic?), but passing along the current CI error:

```
│ Error: Error creating Topic: googleapi: Error 409: Resource already exists in the project (resource=cft-tf-pubsub-topic).
│ 
│   with module.pubsub.google_pubsub_topic.topic[0],
│   on ../../main.tf line 106, in resource "google_pubsub_topic" "topic":
│  106: resource "google_pubsub_topic" "topic" {
│ 
╵}
```

**Comment by rs1986x** [comment]
> t checked if this is related to your change (duplicate pubsub topic?), but passing along the current CI error:

Hi @apeabody - where is the issue? is it applying the example? just to understand where the failure is and help fix it.

File: examples/bigquery/main.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ */
+
+provider "google" {
+  region = "europe-west1"
+}
+
+module "pubsub" {
+  source     = "../../"
+  project_id = var.project_id
+  topic      = "cft-tf-pubsub-topic"
```
Comment (apeabody) [inline examples/bigquery/main.tf:24]: @rs1986x - I think this topic name might be overlapping with the existing simple example.

```suggestion
  topic      = "cft-tf-pubsub-topic-bigquery"
```

Comment (riccardosalamanna) [reply examples/bigquery/main.tf:24]: Makes perfect sense, I did test this one example, not all at the same time... :smile:

**Comment by comment-bot-dev** [comment]
@rs1986x
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by apeabody — APPROVED** [review]
Thanks @rs1986x for the contribution!

**Pull request merged**